ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -579,7 +579,18 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
     -- The major protocol version of the last era is the maximum major protocol
     -- version we support.
     maxMajorProtVer :: MaxMajorProtVer
-    maxMajorProtVer = MaxMajorProtVer (pvMajor protVerBabbage)
+    maxMajorProtVer =
+        nonEmptyLast (exactlyWeakenNonEmpty maxMajorProtVers)
+      where
+        protVers :: Exactly (CardanoShelleyEras StandardCrypto) MaxMajorProtVer
+        protVers = Exactly $
+          -- ensure that these have the same order as 'CardanoShelleyEras'!
```

Comment:
As discussed, a more invasive approach that does not have this (very minor) potential for future incorrect updates would be to modify `ProtocolParams*` to be type-family based and take an `NP` of those as an argument to `protocolInfoCardano`, which the last protocol version can then be derived from.

Reply:
I very much agree. Let's merge this PR now, and open a TechDebt for the follow-up (as it involves at least some `cardano-node` changes etc)